GVFS/GVFS.Common/ProductUpgraderInfo.cs

```diff
@@ -104,5 +129,37 @@ private static void RecursiveDelete(string path)
 
             directory.Delete();
         }
+
+        private static Version ReadHighestRecordedAvailableVersion()
+        {
+            Version highestAvailableVersion = null;
+            string highestAvailableVersionFile = GetHighestAvailableVersionFilePath();
+
+            if (!File.Exists(highestAvailableVersionFile))
+            {
+                return null;
+            }
+
+            string contents = File.ReadAllText(highestAvailableVersionFile).Trim();

```

Comment:
Nit - Do we need to read the whole file? Should we stop at the first non-empty line. (Not a big issue since the highestAvailableVersionFile is not world writable.)

Reply:
Good point - as the file is controlled by us, I didn't stop at the 1st line. If someone is tampering with the file, then we have bigger problems... @wilbaker suggested to not even read the contents of the file, which could also work...

Reply:
If we decide to continue to read the contents of the file I think it's okay to read the entire file.  There should only be one line, and if we read in unexpected data it seems better to have `Version.TryParse` fail than silently ignore the unexpected extra data.